core/src/main/java/com/linecorp/armeria/common/RequestCancellationException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+/**
+ * A {@link RuntimeException} raised when a request is cancelled by the user.
+ */
+public final class RequestCancellationException extends RuntimeException {
+
+    private static final long serialVersionUID = -8891853443874862294L;
+
+    private static final RequestCancellationException INSTANCE = new RequestCancellationException();
+
+    /**
+     * Returns a singleton {@link RequestCancellationException}.
```

Comment:
I just copied over the code from `ResponseTimeoutException`. Is it not sufficient? https://github.com/line/armeria/blob/454a22c9ec52a9267415a112bd2d2f0d82e72932/core/src/main/java/com/linecorp/armeria/client/ResponseTimeoutException.java#L34-L36

Reply:
You have to do this for a singleton, so that the exception becomes immutable:

https://github.com/line/armeria/blob/454a22c9ec52a9267415a112bd2d2f0d82e72932/core/src/main/java/com/linecorp/armeria/server/RequestTimeoutException.java#L29 (Note the boolean parameter)